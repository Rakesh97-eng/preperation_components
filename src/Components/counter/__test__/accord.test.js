import { fireEvent, render, screen } from "@testing-library/react";
import Counter from "../index";
const ReactTestRenderer = require('react-test-renderer');

describe(
    "Counter",()=>{
       it("should increment Counter",()=>{
        render (<Counter/>);
        const counter = screen.getByTestId("counter")
        const incrementBtn = screen.getByTestId("increment");
        expect(counter).toHaveTextContent("4")
        // interact with elements
        const incrementCounter = jest.fn();
        fireEvent.click(incrementBtn);
        expect(incrementCounter).toHaveBeenCalledTimes(0);
        expect(counter).toHaveTextContent("5")
       })
    }
);
describe("Counter Snapshot", () => {
    it("should match DOM Snapshot", () => {
      const tree = ReactTestRenderer.create(<Counter />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  })

  describe('Counter Field',()=>{
    it("Yes it is Password Field",()=>{
        render(<Counter/>);
        const passfield = screen.getByTestId("passfield")
        const checkboxfield = screen.getByTestId("checkbox")
        expect(passfield).toHaveAttribute('type',"text")
        expect(checkboxfield).not.toBeChecked()
    })
    
  })

