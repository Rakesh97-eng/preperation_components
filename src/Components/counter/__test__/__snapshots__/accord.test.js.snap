// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot should match DOM Snapshot 1`] = `
<div
  className="counter"
>
  <button
    data-testid="increment"
    onClick={[Function]}
  >
    +
  </button>
  <p
    data-testid="counter"
  >
    Count: 
    4
  </p>
  <button
    data-testid="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <input
    data-testid="passfield"
    type="text"
  />
  <input
    checked={false}
    data-testid="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
